launchdarkly-android-client-sdk/src/main/java/com/launchdarkly/sdk/android/ContextDataManager.java

```diff
@@ -162,21 +139,21 @@ private void initDataInternal(
             newIndex = index.updateTimestamp(contextId, System.currentTimeMillis())
                     .prune(maxCachedContexts, removedContextIds);
             index = newIndex;
-            flagsContextId = contextId;
-        }
 
-        for (String removedContextId: removedContextIds) {
-            environmentStore.removeContextData(removedContextId);
-            logger.debug("Removed flag data for context {} from persistent store", removedContextId);
-        }
-        if (writeFlagsToPersistentStore && maxCachedContexts != 0) {
-            environmentStore.setContextData(contextId, newData);
-            logger.debug("Updated flag data for context {} in persistent store", contextId);
+            for (String removedContextId: removedContextIds) {
+                environmentStore.removeContextData(removedContextId);
+                logger.debug("Removed flag data for context {} from persistent store", removedContextId);
+            }
+            if (writeFlagsToPersistentStore && maxCachedContexts != 0) {
+                environmentStore.setContextData(contextId, newData);
+                logger.debug("Updated flag data for context {} in persistent store", contextId);
+            }
+            environmentStore.setIndex(newIndex);
```

Comment:
For reviewers:  `environmentStore.setIndex(newIndex);` should be in the lock block.

Reply:
In the long-term I think we should attempt to change this to a queue of operations that apply to the store. Maybe transactional in some way. It would be preferable to not have to lock for the duration of IO.